java-checks/src/main/java/org/sonar/java/checks/NullShouldNotBeUsedWithOptionalCheck.java

```diff
@@ -163,10 +164,21 @@ private static boolean isNull(ExpressionTree expression) {
     return expression.is(Tree.Kind.NULL_LITERAL);
   }
 
-  private void checkNullableAnnotation(ModifiersTree modifiers, String messageFormat) {
-    NullableAnnotationUtils.nullableAnnotation(modifiers)
-      .ifPresent(annotation -> context.reportIssue(this, annotation,
-        String.format(messageFormat, "@" + annotation.annotationType().symbolType().name())));
+  private void checkNullability(Symbol symbol, String messageFormat) {
+    SymbolMetadata.NullabilityLevel level;
+    if (symbol.isVariableSymbol()) {
+      level = SymbolMetadata.NullabilityLevel.VARIABLE;
+    } else {
+      level = SymbolMetadata.NullabilityLevel.METHOD;
+    }
+    SymbolMetadata.NullabilityData nullabilityData = symbol.metadata().nullabilityData();
+    if (nullabilityData.isNullable(level, false, false)) {
```

Comment:
```suggestion
    if (nullabilityData.isNullable(level, true, false)) {
```
When an element is annotated nullable via meta-annotation, we should not report an issue, as this annotation may not be removed.